### ISSUE TITLE
MatchInterPodAffinity

### ISSUE THREAD

**Issue by nixgadget** [spotahome/redis-operator#19]
I seem to be running into the issue,

`No nodes are available that match all of the predicates: MatchInterPodAffinity`

when its scheduling the redis cluster. Any idea what could be causing this ?

**Comment by miles-** [comment]
How many nodes are in your cluster and big did you configure your redis cluster? I'm wondering if this is a case of the number of redis pods being greater than the number of nodes in the cluster.

**Comment by nixgadget** [comment]
Theres 3 nodes in the cluster and i have also specified 3 sentinels and 2 redis pods in config

I also tried with setting `hardAntiAffinity` with no luck

**Comment by ese** [comment]
@nixgadget Thanks for the report. We need more information to debug the problem.
What version of redis-operator and kubernetes are you using? 
could you provide your redisfailover yaml to try reproduce the problem? 
It's weird since 2 redis replicas is a not valid configuration for the operator

**Comment by benjaminjb** [comment]
I'm having a similar problem, but I am trying to run this in a Minikube with one node.

**Setup**
Kubernetes 1.9.2
Minikube 0.24.1
quay.io/spotahome/redis-operator:latest

**Summary**
When I try to deploy a RedisFailover, that first pod (Redis & Sentinel) comes up fine, but then the next three (sentinels) hang in pending with the error: 

`0/1 nodes are available: 1 MatchInterPodAffinity, 1 PodAntiAffinityRulesNotMatch`

"hardAntiAffinity" is set to false in my RedisFailover.yml. I had thought that with "hardAntiAffinity" set to false, it might be able to pack multiple pods into the same node since its only obeying a "preferredDuringSchedulingIgnoredDuringExecution" rule (https://github.com/spotahome/redis-operator/blob/fdd117ddeedff29623e29b78c1025ecbe35711bd/pkg/failover/client.go#L1010-L1043). 

(This would obviously be bad in production for a database like this, but I'm just experimenting with a dev environment.)

**Comment by ese** [comment]
@benjaminjb Sorry I am trying to reproduce in local but I am unable:

**Setup**
Kubernetes 1.9.2
Minikube 0.24.1
quay.io/spotahome/redis-operator:latest

```
apiVersion: spotahome.com/v1alpha1
kind: RedisFailover
metadata:
  name: myredis 
spec:
  hardAntiAffinity: false
```

```
    : kubectl get nodes
NAME      STATUS    ROLES     AGE       VERSION
peweq     Ready     <none>    17m       v1.9.0

    : kubectl get pods
NAME                                         READY     STATUS    RESTARTS   AGE
redisoperator-c68885d9f-gtvhw   1/1       Running   0          10m
rfr-myredis-0                                1/1       Running   0          7m
rfr-myredis-1                                1/1       Running   0          5m
rfr-myredis-2                                1/1       Running   0          3m
rfs-myredis-7775d78956-8wkv6                 1/1       Running   0          8m
rfs-myredis-7775d78956-fkptk                 1/1       Running   0          8m
rfs-myredis-7775d78956-j5sgl                 1/1       Running   0          8m

    : kubectl get redisfailover
NAME      AGE
myredis   3m
```

**Comment by benjaminjb** [comment]
I've reset everything I could to your specs -- 

Minikube v0.25.0
Kubernetes v1.9.0
quay.io/spotahome/redis-operator:latest

-- and I'm still getting that error.

Curiously, when I check the yml in the Minikube dashboard, I found this podAntiAffinity rule:

`{
  "kind": "Deployment",
  "apiVersion": "extensions/v1beta1",
...
  "spec": {
    "template": {
      "spec": {
        "affinity": {
          "podAntiAffinity": {
            "requiredDuringSchedulingIgnoredDuringExecution": [
              {
                "labelSelector": {
                  "matchLabels": {
                    "app": "redis-failover",
                    "component": "sentinel",
                    "sentinel": "myredis"
                  }
                },
                "topologyKey": "kubernetes.io/hostname"
              }
            ]
          }
        },
      }
    },
  },
}`

If I'm passing in hardAntiAffinity as false, why does the deployment have "requiredDuringSchedulingIgnoredDuringExecution" when the `createPodAntiAffinity` function looks like it should have created a "preferredDuringSchedulingIgnoredDuringExecution" rule?

Also curious, if I change from `quay.io/spotahome/redis-operator:latest` to `quay.io/spotahome/redis-operator:0.1.4`, it works perfectly fine. But `0.1.5` and `latest` both fail with this error. Is your `latest` tag definitely pulling the most recent Docker image?

**Comment by ese** [comment]
@benjaminjb its weird since `latest` is mutable maybe you are using an old release. Try to pull or use `0.1.5` tag

**Comment by benjaminjb** [comment]
Trying to be diligent, @ese, I've tried to deploy a bunch of images and this is what I've found.

**Works for me:**
quay.io/spotahome/redis-operator:0.1.0
quay.io/spotahome/redis-operator:0.1.4
quay.io/spotahome/redis-operator:devops-657-add-flag-hard-antiaffinity

**Doesn't work for me:**
quay.io/spotahome/redis-operator:0.1.5
quay.io/spotahome/redis-operator:latest

**Comment by ese** [comment]
@benjaminjb Thanks for your patience and diligince. At the end it was my environment and that feature it was not yet released. 🤦‍♀️ 
I just released a 0.1.6
